sim/core/item_swaps.go

```diff
@@ -8,11 +8,13 @@ import (
 	"github.com/wowsims/cata/sim/core/stats"
 )
 
-type OnItemSwap func(*Simulation, proto.ItemSlot)
+type OnItemSwap func(*Simulation)
+type OnItemSwapSlot func(*Simulation, proto.ItemSlot)
 
 type ItemSwap struct {
-	character       *Character
-	onSwapCallbacks [NumItemSlots][]OnItemSwap
+	character           *Character
+	onSwapCallbacks     []OnItemSwap
+	onSwapSlotCallbacks [NumItemSlots][]OnItemSwapSlot
```

Comment:
Similarly, I would call the first one `onSetSwapCallbacks` and the second one `onSlotSwapCallbacks`.